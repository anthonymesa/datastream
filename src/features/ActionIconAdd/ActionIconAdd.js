import { ActionIcon } from "@mantine/core"
import { FiPlusSquare } from "react-icons/fi"
import { useDispatch } from "react-redux"
import { open, setParentId } from "../ModalActionAdd/ModalActionAddSlice"
import "../ActionEditBar/ActionEditBar.css"

const ActionIconAdd = ({ uuid }) => {

    const dispatch = useDispatch();

    const handleOnClick = () => {
        dispatch(setParentId({ uuid: uuid }))
        dispatch(open({}));
    }

    const props = {
        variant: "subtle",
        className: "EditBarIcon",
        onClick: handleOnClick,
    }

    return (
        <ActionIcon {...props}>
            <FiPlusSquare />
        </ActionIcon>
    )
}

export default ActionIconAdd