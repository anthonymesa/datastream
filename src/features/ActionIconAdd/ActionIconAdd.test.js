import renderer from 'react-test-renderer';
import ActionIconAdd from './ActionIconAdd';

it("Renders an icon", () => {
    const component = renderer.create(
        <ActionIconAdd />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

// it("When clicked, updates state accordingly", () => {

// }) 