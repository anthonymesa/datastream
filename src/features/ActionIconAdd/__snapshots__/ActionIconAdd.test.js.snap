// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders an icon 1`] = `
<button
  className="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-1pt46x6"
  type="button"
>
  <svg
    color="green"
    fill="none"
    height="xl"
    onClick={[Function]}
    radius="xl"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      Object {
        "color": "green",
      }
    }
    variant="filled"
    viewBox="0 0 24 24"
    width="xl"
    xmlns="http://www.w3.org/2000/svg"
  >
    <line
      x1="3"
      x2="21"
      y1="12"
      y2="12"
    />
    <line
      x1="3"
      x2="21"
      y1="6"
      y2="6"
    />
    <line
      x1="3"
      x2="21"
      y1="18"
      y2="18"
    />
  </svg>
</button>
`;
